examples/autoware_diagnostic_graph_test_examples/test/example.cpp

```diff
@@ -0,0 +1,84 @@
+// Copyright 2025 The Autoware Contributors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <ament_index_cpp/get_package_share_directory.hpp>
+#include <autoware/diagnostic_graph_aggregator/tests/tests.hpp>
+
+#include <gtest/gtest.h>
+
+#include <string>
+#include <unordered_map>
+
+using autoware::diagnostic_graph_aggregator::test::DiagnosticArray;
+using autoware::diagnostic_graph_aggregator::test::DiagnosticLevel;
+using autoware::diagnostic_graph_aggregator::test::DiagnosticStatus;
+using autoware::diagnostic_graph_aggregator::test::TestGraph;
+using autoware::diagnostic_graph_aggregator::test::TestNodeStatus;
+
+DiagnosticStatus create_status(const std::string & name, const DiagnosticLevel level)
+{
+  DiagnosticStatus status;
+  status.name = name;
+  status.level = level;
+  return status;
+}
+
+std::unordered_map<std::string, DiagnosticLevel> test(
+  TestGraph & graph, const std::unordered_map<std::string, DiagnosticLevel> & input)
+{
+  DiagnosticArray diags;
+  for (const auto & name : graph.list_diag_names()) {
+    DiagnosticStatus status;
+    status.name = name;
+    status.level = input.count(name) ? input.at(name) : DiagnosticStatus::OK;
+    diags.status.push_back(status);
+  }
+
+  const auto result = graph.test(diags);
+  std::unordered_map<std::string, DiagnosticLevel> output;
+  for (const auto & status : result) {
+    output[status.path] = status.level;
+  }
+  return output;
+}
```

Comment:
Perhaps it might be better to wrap with namespace `autoware::diagnostic_graph_aggregator`

Reply:
For what reason? This code is not referenced by other packages.